# Corporate Financing Decisions

**Definition**: Decisions regarding the sourcing of funds to finance corporate activities.

**Types**:
- **Equity Financing**: Issuing shares to raise capital.
- **Debt Financing**: Borrowing funds through loans or bonds.
- **Hybrid Financing**: Instruments like convertible bonds that have features of both debt and equity.

**Considerations**:
- **Cost of Capital**: The return expected by those who provide capital.
- **Financial Flexibility**: The ability to raise funds on short notice.
- **Control**: Dilution of ownership and control over the company.


# Weighted Average Cost of Capital (WACC)

**Definition**: The average rate of return a company is expected to pay its security holders to finance its assets.

**Formula**:
$ \text{WACC} = \frac{E}{V} \times R_e + \frac{D}{V} \times R_d \times (1 - T_c) $

- $ E $: Market value of equity
- $ D $: Market value of debt
- $ V $: Total market value of equity and debt $ (E + D) $
- $ R_e $: Cost of equity
- $ R_d $: Cost of debt
- $ T_c $: Corporate tax rate

**Purpose**: Assess the cost of financing and make investment decisions.


In [3]:
# Example data
E = 500000   # Market value of equity
D = 200000   # Market value of debt
Re = 0.10    # Cost of equity (10%)
Rd = 0.05    # Cost of debt (5%)
Tc = 0.21    # Corporate tax rate (21%)

# Calculate WACC
V = E + D    # Total market value of equity and debt
WACC = (E / V) * Re + (D / V) * Rd * (1 - Tc)
print(f"WACC: {WACC:.4f}")


WACC: 0.0827


### Explanation of WACC Calculation

The Weighted Average Cost of Capital (WACC) is a crucial metric that helps determine a company's cost of capital, which is the average rate it is expected to pay to all its security holders (debt and equity). In this context:

- **E (Market value of equity)** and **D (Market value of debt)** are the proportions of the company's capital structure.
- **Re (Cost of equity)** is the return required by equity investors.
- **Rd (Cost of debt)** is the effective rate that the company pays on its borrowed funds.
- **Tc (Corporate tax rate)** accounts for the tax shield provided by debt.

The formula used:

$$
\text{WACC} = \left(\frac{E}{V}\right) \times Re + \left(\frac{D}{V}\right) \times Rd \times (1 - Tc)
$$

Where \( V = E + D \).

In our case, with a market value of equity of $500,000, a market value of debt of $200,000, a cost of equity of 10%, a cost of debt of 5%, and a tax rate of 21%, we calculate the WACC as **0.0827**, or **8.27%**.


# Capital Structure and Dividend Policy

- **Capital Structure**: The mix of debt and equity financing used by a company.
- **Trade-Off Theory**: Balances the tax advantages of debt with bankruptcy costs.
- **Pecking Order Theory**: Firms prefer internal financing, then debt, and issue equity as a last resort.
- **Dividend Policy**: Decisions regarding the distribution of profits to shareholders.
- **Residual Dividend Model**: Dividends are based on earnings left after funding all positive NPV projects.
- **Stable Dividend Policy**: Consistent dividends regardless of earnings fluctuations.


# Practical Example: Evaluate the Optimal Capital Structure for a Company

**Scenario**: Evaluate the optimal capital structure for a company.

**Steps**:
1. Analyze the current mix of debt and equity.
2. Consider the trade-offs between tax benefits and bankruptcy costs.
3. Adjust the capital structure to minimize WACC.


In [4]:
import numpy as np

# Example data
debt_ratios = np.arange(0, 1.0, 0.1)  # Exclude 1.0 to avoid division by zero
cost_of_debt = 0.05
cost_of_equity_base = 0.10
tax_rate = 0.21

# Function to calculate WACC for different debt ratios
def calculate_wacc(debt_ratio):
    equity_ratio = 1 - debt_ratio
    if equity_ratio == 0:
        return np.inf  # Return a very large number if division by zero occurs
    cost_of_equity = cost_of_equity_base + (debt_ratio / equity_ratio) * (cost_of_debt - cost_of_equity_base)
    wacc = equity_ratio * cost_of_equity + debt_ratio * cost_of_debt * (1 - tax_rate)
    return wacc

# Calculate WACC for different debt ratios
wacc_values = [calculate_wacc(r) for r in debt_ratios]

# Find the optimal debt ratio
optimal_debt_ratio = debt_ratios[np.argmin(wacc_values)]
print(f"Optimal Debt Ratio: {optimal_debt_ratio:.2f}")
print(f"Minimum WACC: {min(wacc_values):.4f}")



Optimal Debt Ratio: 0.90
Minimum WACC: 0.0006


### Explanation of Optimal Debt Ratio Calculation

To find the optimal capital structure, the Weighted Average Cost of Capital (WACC) is calculated for different debt ratios, ranging from 0% to 100%. The objective is to identify the debt ratio that minimizes the WACC, hence optimizing the company's capital structure.

Key points:

- The **equity ratio** is the complement of the debt ratio (1 - debt ratio).
- The **cost of equity** increases as the debt ratio increases due to the rising financial risk.
- The **optimal debt ratio** is the one that results in the lowest WACC, which indicates the most cost-effective mix of debt and equity financing for the company.

In this scenario, the model calculated the **optimal debt ratio** to be **90%**, corresponding to a **minimum WACC of 0.0006**. This suggests that, under the given assumptions, the company should finance its operations primarily with debt to achieve the lowest cost of capital. However, this result should be interpreted with caution, as extremely high debt levels could increase bankruptcy risk.
